ДЗ по теме "Executors"

daria-grebneva opened the pull request.

daria-grebneva committed:
ДЗ по теме "Executors"

kataus commented on hw15-sequence-of-numbers/src/main/java/SequenceOfNumbersSynchronizer.java:
```diff
@@ -0,0 +1,49 @@
+    }
+
+    private synchronized void task() {
+        lock.lock();
```
и synchronized, и lock'и. Перебор, достаточно одно из механизмов сихнронизации действий

daria-grebneva replied:
Поправлено

daria-grebneva committed:
Правки:
 - Потоки теперь чередуются в выводе
 - Бесконечный цикл вывода значений

kataus commented on hw15-sequence-of-numbers/src/main/java/SequenceOfNumbersSynchronizer.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    private int getIncrementedValue(int result) throws ExecutionException, InterruptedException {
+        return EXECUTOR.submit(() -> increment(result)).get();
```
Порядок старта не гаратнируется.

daria-grebneva replied:
Теперь разобралась с порядком старта)

kataus commented on hw15-sequence-of-numbers/src/main/java/SequenceOfNumbersSynchronizer.java:
```diff
@@ -0,0 +1,78 @@
+
+        var result = new Object() {
+            int value1 = 1;
+            int value2 = 1;
```
Цифра должна быть одна на двоих, общим состоянием

daria-grebneva replied:
Если я правильно понимаю, раз цифра общее состояние, то первый поток выводит - 1, второй - 2, первый -3, второй - 4 и т.д.

daria-grebneva replied:
Поправлено

kataus commented on hw15-sequence-of-numbers/src/main/java/SequenceOfNumbersSynchronizer.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    private int increment(int counter) {
+        LOCK.lock();
```
Если я один поток приторможу (например дебагером) вне локов - то второму это не помешает пойти вперед.

daria-grebneva replied:
Поменяла реализацию

daria-grebneva committed:
Поменяла реализацию
 - Начинает теперь гарантированно первый поток (контролируется с помощью wait)
 - counter - общий для обоих потоков
